packages/features/bookings/lib/handleNewBooking/loadUsers.ts

```diff
@@ -0,0 +1,102 @@
+import { Prisma } from "@prisma/client";
+import type { IncomingMessage } from "http";
+
+import { orgDomainConfig } from "@calcom/features/ee/organizations/lib/orgDomains";
+import { HttpError } from "@calcom/lib/http-error";
+import logger from "@calcom/lib/logger";
+import { UserRepository } from "@calcom/lib/server/repository/user";
+import prisma, { userSelect } from "@calcom/prisma";
+import { credentialForCalendarServiceSelect } from "@calcom/prisma/selects/credential";
+
+import type { NewBookingEventType } from "../handleNewBooking";
+
+const log = logger.getSubLogger({ prefix: ["[loadUsers]:handleNewBooking "] });
+type Hosts = NewBookingEventType["hosts"];
+
+export const loadUsers = async (
+  eventType: NewBookingEventType,
+  dynamicUserList: string[],
+  req: IncomingMessage
+) => {
+  try {
+    return eventType.id
+      ? await loadUsersByEventType(eventType)
+      : await loadDynamicUsers(dynamicUserList, req);
+  } catch (error) {
+    if (error instanceof HttpError || error instanceof Prisma.PrismaClientKnownRequestError) {
+      throw new HttpError({ statusCode: 400, message: error.message });
+    }
+    throw new HttpError({ statusCode: 500, message: "Unable to load users" });
+  }
+};
+
+const loadUsersByEventType = async (
+  eventType: NewBookingEventType
+): Promise<NewBookingEventType["users"]> => {
+  const hosts = eventType.hosts || [];
+  validateHosts(hosts);
+  const users = hosts.map(({ user, isFixed, priority }) => ({
+    ...user,
+    isFixed,
+    priority,
+  }));
+  return users.length ? users : eventType.users;
+};
+
+const validateHosts = (hosts: Hosts): void => {
+  if (!Array.isArray(hosts)) {
+    throw new Error("eventType.hosts is not properly defined.");
+  }
+};
+
+const loadDynamicUsers = async (dynamicUserList: string[], req: IncomingMessage) => {
+  validateDynamicUserList(dynamicUserList);
+  const { isValidOrgDomain, currentOrgDomain } = orgDomainConfig(req);
+  return findUsersByUsername({
+    usernameList: dynamicUserList,
+    orgSlug: isValidOrgDomain ? currentOrgDomain : null,
+  });
+};
+
+const validateDynamicUserList = (dynamicUserList: string[]): void => {
```

Comment:
It isn't doing anything dynamic user related. How about calling it something like `ensureArrayWithAtleastOneItem`? It makes it more usable.

Also, `dynamicUserList` is already being enforced as an array in TypeScript so, we just need to check that it has atleast one entry. Maybe that can be done inline in the calling function itself.